src/Resolvers/AbstractDataResolver.php

```diff
@@ -35,11 +38,34 @@ public function validate(array $data): array {
      * {@inheritDoc}
      */
     final public function resolve(array $data, array $params = []) {
-        $result = $this->validate($data);
-        $result = $this->resolveInternal($result, $params);
+        global $layoutCacheNode;
+        if (!empty($data['$hydrate'])) {
+            $layoutCacheNodeKey = md5(json_encode($data));
```

Comment:
Are we trying to have a unique $layoutCacheNodeKey? The md5 algorithm does not guarantee unique values.

Reply:
This method was requested  by Adam specifically, also the odds of being same are very very low.